src/dynamic_boundary_conditions/hirds_depth_data_from_db.py

```diff
@@ -6,75 +6,75 @@
+    if duration != "all":
+        rain_depth = rain_depth[["site_id", "rcp", "time_period", "ari", "aep", f"{duration}"]]
+    return rain_depth
+
+
+def get_each_site_rain_depth_data(
+        engine, site_id: str, rcp: float, time_period: str, ari: float, duration: str) -> pd.DataFrame:
+    """Get the hirds rainfall data for the requested site from the database and return in dataframe format."""
+    if (rcp is None and time_period is not None) or (rcp is not None and time_period is None):
+        log.error(
+            "Check the arguments of the 'rain_depths_from_db' function. "
+            "If rcp is None, time period should be None, and vice-versa.")
+        sys.exit()
     elif rcp is not None and time_period is not None:
-        query = f"""select site_id, "{duration}h" from hirds_rain_depth where
-                site_id='{site}' and ari={ari} and\
-                rcp='{rcp}' and time_period='{time_period}'"""
+        query = f"""SELECT * FROM rainfall_depth
+        WHERE site_id='{site_id}' AND rcp='{rcp}' AND time_period='{time_period}' AND ari={ari};"""
+        rain_depth = pd.read_sql_query(query, engine)
     else:
-        query = f"""select site_id, "{duration}h" from hirds_rain_depth where
-                site_id='{site}' and ari={ari} and\
-                rcp is null and time_period is null"""
-    rain_depth = engine.execute(query)
-    rain_depth = list(rain_depth.fetchone())
+        query = f"""SELECT * FROM rainfall_depth
+        WHERE site_id='{site_id}' AND rcp IS NULL AND time_period IS NULL AND ari={ari};"""
+        rain_depth = pd.read_sql_query(query, engine).head(1)
+    rain_depth = filter_for_duration(rain_depth, duration)
     return rain_depth
 
 
-def hirds_depths_from_db(
-    engine, catchment_area, ari, duration, rcp=None, time_period=None
-):
-    """Get the list of depths and site's id of each site and return in
-    dataframe format."""
-    sites_in_catchment = hirds_depth_data_to_db.get_sites_in_catchment(
-        catchment_area, engine
-    )
+def rain_depths_from_db(
+        engine, catchment_polygon: Polygon, rcp: float, time_period: str, ari: float, duration: str) -> pd.DataFrame:
+    """Get all the rainfall data for the sites within the catchment area and return in dataframe format."""
+    sites_id_in_catchment = hirds_depth_data_to_db.get_sites_id_in_catchment(engine, catchment_polygon)
 
-    depths_list = []
-    for site_id in sites_in_catchment:
-        rain_depth = get_each_site_hirds_depth_data(
-            ari, duration, site_id, engine, rcp, time_period
-        )
-        depths_list.append(rain_depth)
-    rain_depth_data = pd.DataFrame(depths_list, columns=["site_id", "depth"])
-    return rain_depth_data
+    rain_depth_in_catchment = pd.DataFrame()
+    for site_id in sites_id_in_catchment:
+        rain_depth = get_each_site_rain_depth_data(engine, site_id, rcp, time_period, ari, duration)
```

Comment:
Would be good to have a convo with you about how we could optimise this using more SQL and less Python. Lets have a 5 min catchup and see if its worth it or not worth the hassle @caseyli92

Reply:
As discussed, the generation of output is pretty fast, thus not worth the hassle :)